pkgs/development/python-modules/keyutils/default.nix

```diff
@@ -12,6 +12,10 @@ buildPythonPackage rec {
     sha256 = "0pfqfr5xqgsqkxzrmj8xl2glyl4nbq0irs0k6ik7iy3gd3mxf5g1";
   };
 
+  postPatch = ''
+    substituteInPlace setup.py --replace '"pytest-runner"' ""
+  '';
+
   buildInputs = [ keyutils ];
   checkInputs = [ pytest pytest-runner ];
```

Comment:
Runner can be removed as it's not needed

Reply:
```
Executing setuptoolsCheckPhase
usage: nix_run_setup [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
   or: nix_run_setup --help [cmd1 cmd2 ...]
   or: nix_run_setup --help-commands
   or: nix_run_setup cmd --help

error: invalid command 'pytest'
```

Reply:
forgot that it needs pytestCheckHook.
but it looks like pytestCheckHook is broken for this for some reason (even if pytest-runner is left in checkInputs)
```diff
diff --git a/pkgs/development/python-modules/keyutils/default.nix b/pkgs/development/python-modules/keyutils/default.nix
index 21eeb11d136..cfabae3a3b4 100644
--- a/pkgs/development/python-modules/keyutils/default.nix
+++ b/pkgs/development/python-modules/keyutils/default.nix
@@ -1,4 +1,4 @@
-{ lib, buildPythonPackage, fetchFromGitHub, keyutils, pytest-runner, pytest }:
+{ lib, buildPythonPackage, fetchFromGitHub, keyutils, pytestCheckHook }:
 
 buildPythonPackage rec {
   pname = "keyutils";
@@ -17,7 +17,7 @@ buildPythonPackage rec {
   '';
 
   buildInputs = [ keyutils ];
-  checkInputs = [ pytest pytest-runner ];
+  checkInputs = [ pytestCheckHook ];
 
   meta = {
     description = "A set of python bindings for keyutils";
```

```
python3.9-keyutils> ============================= test session starts ==============================
python3.9-keyutils> platform linux -- Python 3.9.6, pytest-6.2.5, py-1.10.0, pluggy-1.0.0
python3.9-keyutils> rootdir: /build/source, configfile: setup.cfg
python3.9-keyutils> collected 0 items / 1 error
python3.9-keyutils> ==================================== ERRORS ====================================
python3.9-keyutils> ____________________ ERROR collecting test/keyutils_test.py ____________________
python3.9-keyutils> ImportError while importing test module '/build/source/test/keyutils_test.py'.
python3.9-keyutils> Hint: make sure your test modules/packages have valid Python names.
python3.9-keyutils> Traceback:
python3.9-keyutils> /nix/store/8dxxjbiyxwkvh53q5kh6nydla2anacgi-python3-3.9.6/lib/python3.9/importlib/__init__.py:127: in import_module
python3.9-keyutils>     return _bootstrap._gcd_import(name[level:], package, level)
python3.9-keyutils> test/keyutils_test.py:23: in <module>
python3.9-keyutils>     import keyutils
python3.9-keyutils> keyutils/__init__.py:21: in <module>
python3.9-keyutils>     from . import _keyutils
python3.9-keyutils> E   ImportError: cannot import name '_keyutils' from partially initialized module 'keyutils' (most likely due to a circular import) (/build/source/keyutils/__init__.py)
python3.9-keyutils> =========================== short test summary info ============================
python3.9-keyutils> ERROR test/keyutils_test.py
python3.9-keyutils> !!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
python3.9-keyutils> =============================== 1 error in 0.06s ===============================
```

Reply:
https://stackoverflow.com/a/66055075
`For future readers, this can also happen if you name a python file the same name as a dependency your project uses.`

Reply:
So I should leave it as-is, right?

Reply:
Yes